test/suites/config.sh

```diff
@@ -404,38 +407,39 @@ test_container_metadata() {
 }
 
 test_container_snapshot_config() {
+    local unbuffer=""
     if ! tty -s; then
-        echo "==> SKIP: test_container_snapshot_config requires a terminal"
-        return
+        unbuffer="$(command -v unbuffer)"
+        if [ -z "${unbuffer}" ]; then
+            echo "==> SKIP: test_container_snapshot_config requires a terminal or 'unbuffer'"
+            return
+        fi
     fi
 
-    ensure_import_testimage
-
-    lxc init testimage foo -s "lxdtest-$(basename "${LXD_DIR}")"
+    lxc init --empty foo
     lxc snapshot foo
-    lxc config show foo/snap0 | grep -F 'expires_at: 0001-01-01T00:00:00Z'
+    [ "$(lxc config get foo/snap0 expires_at --property)" = "0001-01-01 00:00:00 +0000 UTC" ]
 
     echo 'expires_at: 2100-01-01T00:00:00Z' | lxc config edit foo/snap0
-    lxc config show foo/snap0 | grep -F 'expires_at: 2100-01-01T00:00:00Z'
+    [ "$(lxc config get foo/snap0 expires_at --property)" = "2100-01-01 00:00:00 +0000 UTC" ]
 
     # Remove expiry date using zero time
     echo 'expires_at: 0001-01-01T00:00:00Z' | lxc config edit foo/snap0
-    lxc config show foo/snap0 | grep -F 'expires_at: 0001-01-01T00:00:00Z'
+    [ "$(lxc config get foo/snap0 expires_at --property)" = "0001-01-01 00:00:00 +0000 UTC" ]
 
     echo 'expires_at: 2100-01-01T00:00:00Z' | lxc config edit foo/snap0
-    lxc config show foo/snap0 | grep -F 'expires_at: 2100-01-01T00:00:00Z'
+    [ "$(lxc config get foo/snap0 expires_at --property)" = "2100-01-01 00:00:00 +0000 UTC" ]
 
     # Remove expiry date using empty value
     echo 'expires_at:' | lxc config edit foo/snap0
-    lxc config show foo/snap0 | grep -F 'expires_at: 0001-01-01T00:00:00Z'
+    [ "$(lxc config get foo/snap0 expires_at --property)" = "0001-01-01 00:00:00 +0000 UTC" ]
 
     # Check instance name is included in edit screen.
-    cmd=$(unset -f lxc; command -v lxc)
-    output=$(EDITOR="cat" timeout --foreground 120 "${cmd}" config edit foo/snap0)
-    echo "${output}" | grep "name: snap0"
+    output=$(EDITOR="cat" ${unbuffer} timeout --foreground 120 "${_LXC}" config edit foo/snap0)
+    echo "${output}" | grep -xF "name: snap0"
 
     # Check expanded config isn't included in edit screen.
-    ! echo "${output}"  | grep "expanded" || false
+    ! echo "${output}" | grep -F "expanded" || false
```

Comment:
For me it's preferable to never try to grep for something that doesn't exist. Can we use `yq` here to make assertions like we do with `jq`? Not sure if it supports it.

Reply:
If you don't mind, I'll leave that one for another PR.